engine-standalone-storage/src/lib.rs

```diff
@@ -0,0 +1,211 @@
+        self.db.put(storage_key, block_height_bytes)
+    }
+
+    pub fn get_transaction_by_hash(
+        &self,
+        tx_hash: H256,
+    ) -> Result<TransactionIncluded, error::Error> {
+        let storage_key =
+            construct_storage_key(StoragePrefix::TransactionPosition, tx_hash.as_ref());
+        self.db
+            .get_pinned(storage_key)?
+            .map(|slice| {
+                let mut buf = [0u8; 34];
+                buf.copy_from_slice(slice.as_ref());
+                TransactionIncluded::from_bytes(buf)
+            })
+            .ok_or(error::Error::TransactionHashNotFound(tx_hash))
+    }
+
+    pub fn get_transaction_by_position(
+        &self,
+        tx_included: TransactionIncluded,
+    ) -> Result<H256, error::Error> {
+        let storage_key =
+            construct_storage_key(StoragePrefix::TransactionHash, &tx_included.to_bytes());
+        self.db
+            .get_pinned(storage_key)?
+            .map(|slice| H256::from_slice(slice.as_ref()))
+            .ok_or(error::Error::TransactionNotFound(tx_included))
+    }
+
+    pub fn get_transaction_diff(
+        &self,
+        tx_included: TransactionIncluded,
+    ) -> Result<Diff, error::Error> {
+        let storage_key = construct_storage_key(StoragePrefix::Diff, &tx_included.to_bytes());
+        self.db
+            .get_pinned(storage_key)?
+            .map(|slice| Diff::try_from_bytes(slice.as_ref()).unwrap())
+            .ok_or(error::Error::TransactionNotFound(tx_included))
+    }
+
+    pub fn set_transaction_included(
+        &mut self,
+        tx_hash: H256,
+        tx_included: &TransactionIncluded,
+        diff: &Diff,
+    ) -> Result<(), error::Error> {
+        let tx_included_bytes = tx_included.to_bytes();
+
+        let storage_key = construct_storage_key(StoragePrefix::TransactionHash, &tx_included_bytes);
+        self.db.put(storage_key, tx_hash)?;
+
+        let storage_key =
+            construct_storage_key(StoragePrefix::TransactionPosition, tx_hash.as_ref());
+        self.db.put(storage_key, tx_included_bytes)?;
+
+        let storage_key = construct_storage_key(StoragePrefix::Diff, &tx_included_bytes);
+        self.db.put(storage_key, diff.try_to_bytes().unwrap())?;
+
+        let block_height = self.get_block_height_by_hash(tx_included.block_hash)?;
+        for (key, value) in diff.iter() {
+            let storage_key = construct_engine_key(key, block_height, tx_included.position);
+            self.db.put(storage_key, value.try_to_bytes().unwrap())?;
+        }
+
+        Ok(())
+    }
+
+    /// Get an object which represents the state of the engine at the given block hash,
+    /// after transactions up to (not including) the given transaction index.
+    /// The `input` is the bytes that would be present in the NEAR runtime (normally
+    /// not needed for standlone engine).
+    pub fn get_engine_storage<'db, 'input: 'db>(
```

Comment:
`take_engine_storage_at_position` would be a better choice here, given what is happening in the method.

Reply:
I don't agree with the `take` verb. I think of `take` as consuming `self`, where as this only requires a reference `&mut self`. Granted the `get` verb is maybe not quite right either because the returned object can mutate the underlying storage. How about we call it `access_engine_storage_at_position`?

Reply:
Yeah, I like `acess_engine...` better. Lets go with that. I don't like using `get` as in Rust that heavily implies that it is taking itself as a reference, not a mut reference.

Reply:
Done in https://github.com/aurora-is-near/aurora-engine/pull/375/commits/01d802677043ec56ccf13ef9ed34c6f739c2e799